### PR TITLE
fix for caldo up t wifi

### PR DESCRIPTION
**Pull request by sebastianciupinski**

### CODE REVIEW

**Comment by make-all** [comment]
I don't see how this is an improvement

**Comment by make-all** [comment]
I see you have changed the dps_val for fan speed, but can you provide a local log to show one of these values? Because if it is just taken from the iot.tuya.com developer portal, there can be differences between cloud and local values like this.  I have to assume the originally supplied values were correct in absence of any evidence.

**Comment by sebastianciupinski** [comment]
@make-all this is improvement in a way that previous version simply did not work with this device.

I was originally adding this device in PR 1296, but after modification you did and that were merged it did not work.

This PR is fixing that.

**Comment by sebastianciupinski** [comment]
`{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2024.7.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.12.4",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Oslo",
    "os_name": "Linux",
    "os_version": "6.6.33-haos",
    "supervisor": "2024.06.2",
    "host_os": "Home Assistant OS 12.4",
    "docker_version": "26.1.4",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "tuya_local": {
      "documentation": "https://github.com/make-all/tuya-local",
      "version": "2024.6.0",
      "requirements": [
        "tinytuya==1.14.0",
        "tuya-device-sharing-sdk==0.1.9"
      ]
    },
    "virtual": {
      "documentation": "https://github.com/twrecked/hass-virtual/blob/master/README.md",
      "version": "0.8.0.1",
      "requirements": []
    },
    "webrtc": {
      "documentation": "https://github.com/AlexxIT/WebRTC",
      "version": "v3.5.2",
      "requirements": []
    },
    "spook": {
      "documentation": "https://spook.boo",
      "version": "3.0.1",
      "requirements": []
    },
    "generic_water_heater": {
      "documentation": "https://www.home-assistant.io/integrations/generic_water_heater",
      "version": "0.0.3",
      "requirements": []
    },
    "scheduler": {
      "documentation": "https://github.com/nielsfaber/scheduler-component",
      "version": "v0.0.0",
      "requirements": []
    },
    "auto_backup": {
      "documentation": "https://github.com/jcwillox/hass-auto-backup",
      "version": "1.5.0",
      "requirements": []
    },
    "tuya_cloud_map_extractor": {
      "documentation": "https://github.com/oven-lab/tuya_cloud_map_extractor/blob/main/README.md",
      "version": "v1.2.0",
      "requirements": [
        "numpy",
        "pillow",
        "requests"
      ]
    },
    "smartthinq_sensors": {
      "documentation": "https://github.com/ollo69/ha-smartthinq-sensors",
      "version": "0.39.2",
      "requirements": [
        "pycountry>=23.12.11",
        "xmltodict>=0.13.0",
        "charset_normalizer>=3.2.0"
      ]
    },
    "dual_smart_thermostat": {
      "documentation": "https://github.com/swingerman/ha-dual-smart-thermostat.git",
      "version": "0.9.6",
      "requirements": []
    },
    "nordpool": {
      "documentation": "https://github.com/custom-components/nordpool/",
      "version": "0.0.14",
      "requirements": [
        "nordpool>=0.2",
        "backoff"
      ]
    },
    "spook_inverse": {
      "documentation": "https://spook.boo",
      "version": "3.0.1",
      "requirements": []
    },
    "hacs": {
      "documentation": "https://hacs.xyz/docs/configuration/start",
      "version": "1.34.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "xiaomi_cloud_map_extractor": {
      "documentation": "https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor",
      "version": "v2.2.0",
      "requirements": [
        "pillow",
        "pybase64",
        "python-miio",
        "requests",
        "pycryptodome"
      ]
    }
  },
  "integration_manifest": {
    "domain": "tuya_local",
    "name": "Tuya Local",
    "codeowners": [
      "make-all"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/make-all/tuya-local",
    "integration_type": "device",
    "iot_class": "local_push",
    "issue_tracker": "https://github.com/make-all/tuya-local/issues",
    "requirements": [
      "tinytuya==1.14.0",
      "tuya-device-sharing-sdk==0.1.9"
    ],
    "version": "2024.6.0",
    "is_built_in": false
  },
  "setup_times": {
    "null": {
      "setup": 2.9921066015958786e-05
    },
    "029b21496f4306cd442426b8554bec0f": {
      "wait_import_platforms": -0.0046515329740941525,
      "config_entry_setup": 1.519773957086727
    },
    "b54c7a2d20610ffa87373082fa960eed": {
      "wait_import_platforms": -0.0014874970074743032,
      "config_entry_setup": 4.916927313897759
    }
  },
  "data": {
    "name": "CALDO UP T WIFI",
    "type": "caldo_up_t_wifi",
    "device_id": "**REDACTED**",
    "device_cid": "",
    "local_key": "**REDACTED**",
    "host": "**REDACTED**",
    "protocol_version": 3.3,
    "tinytuya_version": "1.14.0",
    "api_version_set": 3.3,
    "api_version_used": 3.3,
    "api_working": true,
    "status": {},
    "cached_state": {
      "updated_at": 1721862449.7448854,
      "1": false,
      "2": 18,
      "3": 23,
      "5": "level_1",
      "8": false,
      "19": "cancel",
      "101": false
    },
    "pending_state": {},
    "connected": true,
    "force_dps": [],
    "home_assistant": {
      "name": "CALDO UP T WIFI",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "climate.caldo_up_t_wifi_2",
            "state": "off",
            "attributes": {
              "hvac_modes": [
                "heat_cool",
                "off"
              ],
              "min_temp": 18,
              "max_temp": 45,
              "target_temp_step": 1.0,
              "preset_modes": [
                "fan only",
                "comfort",
                "boost"
              ],
              "swing_modes": [
                "off",
                "on"
              ],
              "current_temperature": 23,
              "temperature": 18,
              "preset_mode": "fan only",
              "swing_mode": "off",
              "friendly_name": "CALDO UP T WIFI",
              "supported_features": 433
            },
            "last_changed": "2024-07-24T18:58:29.918893+00:00",
            "last_reported": "2024-07-24T22:25:22.664099+00:00",
            "last_updated": "2024-07-24T22:25:22.664099+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "switch.caldo_up_t_wifi_window_detection",
            "state": "off",
            "attributes": {
              "icon": "mdi:window-open-variant",
              "friendly_name": "CALDO UP T WIFI Window detection"
            },
            "last_changed": "2024-07-24T18:58:29.918115+00:00",
            "last_reported": "2024-07-24T18:58:29.918115+00:00",
            "last_updated": "2024-07-24T18:58:29.918115+00:00"
          }
        }
      ]
    }
  }
}`

File: custom_components/tuya_local/devices/caldo_up_t_wifi.yaml [hunk 1]
```diff
@@ -4,20 +4,18 @@ products:
...
   entity: climate
-  translation_only_key: heater
   dps:
     - id: 1
       type: boolean
       name: hvac_mode
       mapping:
         - dps_val: true
-          value: heat
+          value: "heat_cool"
```
Comment (make-all) [inline custom_components/tuya_local/devices/caldo_up_t_wifi.yaml:13]: This is a heater, there is no reason to change this from heat to heat_cool, as it is not capable of automatically switching between heating and cooling.

File: custom_components/tuya_local/devices/caldo_up_t_wifi.yaml [hunk 2]
```diff
@@ -4,20 +4,18 @@ products:
     name: Olimpia Splendid Caldo Up T
 primary_entity:
   entity: climate
-  translation_only_key: heater
   dps:
```
Comment (make-all) [inline custom_components/tuya_local/devices/caldo_up_t_wifi.yaml:7]: There is no good reason to remove the translation_key.

File: custom_components/tuya_local/devices/caldo_up_t_wifi.yaml [hunk 3]
```diff
@@ -4,20 +4,18 @@ products:
...
         - dps_val: true
-          value: heat
+          value: "heat_cool"
         - dps_val: false
           value: "off"
     - id: 2
       type: integer
       name: temperature
-      unit: C
       range:
```
Comment (make-all) [inline custom_components/tuya_local/devices/caldo_up_t_wifi.yaml:19]: Removing the unit is not just without reason, it breaks the integration on HA 2024.6 and later, which requires the unit to be set.

File: custom_components/tuya_local/devices/caldo_up_t_wifi.yaml [hunk 4]
```diff
@@ -28,11 +26,11 @@ primary_entity:
       type: string
       name: preset_mode
       mapping:
-        - dps_val: "0"
-          value: none
-        - dps_val: "1"
+        - dps_val: "level_1"
+          value: "fan only"
```
Comment (make-all) [inline custom_components/tuya_local/devices/caldo_up_t_wifi.yaml:30]: fan_only is not a standard preset name. I think this is better left as "none", which conveys that compared to "comfort" and "boost", it provides no heating.

**Pull request closed without merging**